### PR TITLE
fix(select/dropdown): keyboard ctrl

### PR DESCRIPTION
**Pull request by hchevallier**
# General summary
The keyboard controls were broken on the Select and Dropdown component. 
What this PR does : 
- Add a `useFocusOnOpen` that put the focus on a given element when the open is set to `true`. 
- If the direct child of a `Dropdown` component is a `List` inject a ref to focus it when opening the drodpown
- Fix some lints and types.

# Screenshots

# Check list

-   [ ] (if has style) Add `need: review-integration` label
-   [ ] (if has textual documentation) Add `need: review-doc` label
-   [x] (if has code) Add `need: review-frontend` label
-   [x] (if has react) Check through the [react dev check list]
-   [x] (if fix or feature) Add `need: test` label

Check out the [contribution guide] for general guidelines for submissions to the design system.

[contribution guide]: https://github.com/lumapps/design-system/blob/master/CONTRIBUTING.md
[react dev check list]: https://github.com/lumapps/design-system/blob/master/CONTRIBUTING.md#react-developpment-check-list

### CODE REVIEW

File: packages/lumx-react/src/components/autocomplete/__snapshots__/Autocomplete.test.tsx.snap [hunk 1]
```diff
@@ -94,6 +95,7 @@ exports[`<Autocomplete> Props should render correctly when the dropdown is close
           Montevideo
         </div>
       </ListItem>
+      ,
```
Comment (gcornut) [inline packages/lumx-react/src/components/autocomplete/__snapshots__/Autocomplete.test.tsx.snap]: Booh

File: packages/lumx-react/src/hooks/useChipGroupNavigation.tsx [hunk 2]
```diff
@@ -28,8 +28,8 @@ const INITIAL_STATE_ACTIVE_CHIP = -1;
  * @return chip navigation tools.
  */
 const useChipGroupNavigation: useChipGroupNavigationType = (
-    chips: [],
-    onChipDeleted: (chipDeleted: object) => {},
+    chips: object[],
+    onChipDeleted: (chipDeleted: object) => void,
```
Comment (gcornut) [inline packages/lumx-react/src/hooks/useChipGroupNavigation.tsx]: You can remove the type here

File: packages/lumx-react/src/hooks/useFocusOnOpen.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * Focus the element when the select is opene.
+ *
+ * @param element             Element to focus.
+ * @param isOpen              Whether or not the select is open.
+ * @param shouldFocusOnOpen   Whether or not the focus should be set.
+ */
+function useFocusOnOpen(
+    element: HTMLElement | null,
+    isOpen: boolean | undefined,
+    shouldFocusOnOpen: boolean | undefined,
```
Comment (gcornut) [inline packages/lumx-react/src/hooks/useFocusOnOpen.tsx]: Can we remove null and undefined type here?

File: packages/lumx-react/src/hooks/useKeyboardListNavigation.tsx [hunk 4]
```diff
@@ -52,10 +52,10 @@ const INITIAL_INDEX = -1;
...
-    onListItemNavigated?: (itemSelected: object) => {},
-    onEnterPressed?: (itemSelected: object) => {},
-    onBackspacePressed?: (evt: KeyboardEvent) => {},
+    onListItemSelected?: (itemSelected: object) => void,
+    onListItemNavigated?: (itemSelected: object) => void,
+    onEnterPressed?: (itemSelected: object) => void,
+    onBackspacePressed?: (evt: KeyboardEvent) => void,
     keepFocusAfterSelection: boolean = false,
     initialIndex: number = INITIAL_INDEX,
     preventTabOnEnteredValue: boolean = true,
```
Comment (gcornut) [inline packages/lumx-react/src/hooks/useKeyboardListNavigation.tsx]: Same here, you whould be able to remove these types

**Comment by matmkian** [comment]
Test ko on dropdown (on demo site): navigate to a button with tab -> navigate through items in the list (OK) -> tap enter or ecape doesn't put back the focus on the button

**Comment by matmkian** [comment]
Also on select component, please add arrow down key press to open dropdown when focused on the input (same as angularjs). Today, only enter key press triggers the dropdown.

File: packages/lumx-react/src/components/select/Select.tsx [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
 import { mdiAlertCircle, mdiCheckCircle, mdiClose, mdiCloseCircle, mdiMenuDown } from '@lumx/icons';
 
-import { Chip, ChipGroup, Dropdown, Emphasis, Icon, IconButton, Placement, Size, Theme } from '@lumx/react';
+import { Emphasis, Size, Theme } from '@lumx/react/components';
+import { IconButton } from '@lumx/react/components/button/IconButton';
+import { Chip } from '@lumx/react/components/chip/Chip';
+import { ChipGroup } from '@lumx/react/components/chip/ChipGroup';
+import { Dropdown } from '@lumx/react/components/dropdown/Dropdown';
+import { Icon } from '@lumx/react/components/icon/Icon';
+import { Placement } from '@lumx/react/components/popover/Popover';
```
Comment (pidupuis) [inline packages/lumx-react/src/components/select/Select.tsx]: Is there any reason to use this import style instead of the previous one?

Comment (hchevallier) [reply packages/lumx-react/src/components/select/Select.tsx]: For an unknown reason my tests were completely broken with the previous one

**Pull request merged**